### PR TITLE
Add task breakdown for Smithy orders `.smithy/` template support

### PR DESCRIPTION
**Pull request by JMBattista**
## Summary
- **Primary outcome**: Document the implementation tasks for adding `.smithy/` template resolution and interpolation to the orders command, split into two reviewable slices.
- **Notable behaviour changes**: None yet — this is a task specification document that will guide implementation.
- **Follow-up work deferred**: Implementation of Slice 1 and Slice 2 tasks; integration testing once code changes are in place.

## Context
This task document operationalizes [User Story 2](specs/2026-03-21-002-smithy-orders-issue-templates/smithy-orders-issue-templates.spec.md) from the Smithy orders issue templates specification. The spec defines the requirement for orders to resolve issue body templates from `.smithy/` directories with built-in fallback and variable interpolation. This document breaks that requirement into two implementation slices with clear acceptance criteria and dependencies.

**Why now**: The spec is finalized and ready for implementation. Breaking it into slices allows Slice 1 (template resolution algorithm and variable definitions) to be reviewed and tested independently before Slice 2 (integration into the ticket creation pipeline) is implemented.

**User story unlocked**: Enables users to customize issue body templates per repository without modifying the core orders prompt, addressing FR-005, FR-006, FR-007, FR-008, and FR-010.

### Slice Structure
- **Slice 1** (Template Resolution): Adds a new "Phase 3b: Resolve Body Template" section to `src/templates/base/smithy.orders.md` with:
  - Resolution algorithm (check `.smithy/<type>.md`, fall back to built-in)
  - Interpolation context tables for all 4 artifact types (rfc, features, spec, tasks)
  - Built-in default templates as fallback
  - `{{next_step}}` mapping (rfc→render, features→mark, spec→cut, tasks→forge)

- **Slice 2** (Integration): Restructures the "Create Tickets" phase to use the template pipeline:
  - Replace hardcoded body text with template resolution + interpolation
  - Update rules section with template usage guidance
  - Add tests verifying prompt structure and variable definitions

### Key Architectural Choices
- **Additive approach**: Slice 1 is self-contained and doesn't modify existing phases — the new phase is inserted between current Phase 3 and Phase 4, with subsequent phases renumbered. This minimizes risk and allows independent review.
- **Variable naming**: Uses names from the data model spec (`{{title}}`, `{{next_step}}`, `{{parent_issue}}`, plus type-specific variables) for consistency.
- **Fallback strategy**: Empty `.smithy/<type>.md` files are valid overrides (title set independently), allowing users to suppress default body content if needed.
- **Malformed placeholder handling**: Unknown or malformed placeholders (`{{}}`, `{{ }}`) are left as literal text rather than erroring, making templates more resilient.

### Impacted Modules
- `src/templates/base/smithy.orders.md` — the orders command prompt (primary change location)
- Test files — new tests to verify prompt structure and variable definitions

## Risks & Mitigations

| Risk | Mitigation |
|------|-----------|
| Phase renumbering (old Phase 4 → Phase 5, etc.) could confuse existing references to the prompt | Update all internal references and documentation; verify no external docs reference phase numbers by name. |
| Template resolution algorithm complexity could be misunderstood by the agent | Document the algorithm step-by-step with clear examples in the prompt; include the built-in defaults as reference. |
| Variable interpolation could fail silently if context is incomplete | Specify that unknown variables are left as literal text; document which variables are required vs. optional per artifact type. |
| Tests may not catch all edge cases (empty templates, missing variables, etc.) | Add tests for: (a) prompt contains resolution phase, (b) prompt references `.smithy/` directory, (c) interpolation context tables present for all types, (d) `{{next_step}}` mapping documented. |

## Rollback Plan
- This is a task specification document (`.md` file), not executable code.
-

https://claude.ai/code/session_01NwPUcF4JeRjZGcwK3tGDZ3

### CODE REVIEW

File: specs/2026-03-21-002-smithy-orders-issue-templates/02-orders-uses-smithy-templates.tasks.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+**Justification**: This slice completes the integration — the agent now follows the resolve → build context → interpolate → write flow for every ticket it creates. Without this, the resolution phase from Slice 1 is documentation only.
+
+**Addresses**: FR-005, FR-007, FR-008, FR-010; Acceptance Scenarios 1–6
+
+### Tasks
+
+- [ ] Restructure the "Create Tickets" phase in `src/templates/base/smithy.orders.md` to follow a 3-step pipeline for each ticket: (1) resolve the body template (custom or built-in default from the new phase), (2) build the interpolation context from the parsed artifact data, (3) interpolate `{{variable}}` placeholders and write the result to the temp body file.
+- [ ] Replace the 4 hardcoded `cat > /tmp/orders_body.md << 'BODY' ... BODY` blocks (rfc parent, rfc child, features child, spec child, tasks child) with instructions that reference the resolved template and interpolation context. Keep the `gh issue create --title ... --body-file` pattern unchanged.
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply specs/2026-03-21-002-smithy-orders-issue-templates/02-orders-uses-smithy-templates.tasks.md]: Fixed — updated count from 4 to 5. The RFC parent block was missed in the original count.

---
_Generated by [Claude Code](https://claude.ai/code)_
